src/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -50,15 +50,74 @@ public enum GCNotificationStatus
         NotApplicable = 4
     }
 
+    public struct GCMemoryInfo
+    {
+        /// <summary>
+        /// HighMemoryLoadThreshold when the last GC occured
+        /// </summary>
+        public int HighMemoryLoadThreshold { get; }
+
+        /// <summary>
+        /// Total available memory for the GC to use when the last GC ocurred. By default this is the physical memory on the machine, but it may be customized by specifying a HardLimit.
+        /// </summary>
+        public long TotalAvailableMemory { get; }
+
+        /// <summary>
+        /// Memory Load when the last GC ocurred
+        /// </summary>
+        public int MemoryLoad { get; }
+
+        /// <summary>
+        /// The total heap size when the last GC ocurred
+        /// </summary>
+        public IntPtr HeapSize { get; }
+
+        /// <summary>
+        /// The total fragmentation of the last GC ocurred
+        /// </summary>
+        public IntPtr Fragmentation { get; } // should we include "Bytes" in all names where appropriate?
```

Comment:
Fragmentation doesn't make sense as an IntPtr either. It should either be a long (if its representing bytes) or an int if it represents a percentage.